### PR TITLE
[WIP] adding artifact listing based on tags

### PR DESCRIPTION
**Pull request by sbnetops**

### CODE REVIEW

File: src/polyswarm_api/api.py [hunk 1]
```diff
@@ -349,6 +349,15 @@ def tag_link_update(self, sha256, tags=None, families=None, remove=False):
         """
         return self.generator.update_tag_link(sha256, tags=tags, families=families, remove=remove).execute().result
 
+    def tag_link_list(self, tags=None, families=None, or_tags=None, or_families=None):
+        """
+        Fetch all existing TagLinks for the provided tags.
+        :param tags: A list of tags we want links for.
+        :return: A TagLink resource
```
Comment (mrtizmoatwork) [inline src/polyswarm_api/api.py]: missing the other params in the docs here. What are they, what do they mean?

File: src/polyswarm_api/api.py [hunk 2]
```diff
@@ -349,6 +349,18 @@ def tag_link_update(self, sha256, tags=None, families=None, remove=False):
         """
         return self.generator.update_tag_link(sha256, tags=tags, families=families, remove=remove).execute().result
 
+    def tag_link_list(self, tags=None, families=None, or_tags=None, or_families=None):
+        """
+        Fetch all existing TagLinks for the provided tags.
+        :param tags: A list of tags that must be associated with the TagLinks listed.
+        :param families: A list of families that must be associated with the TagLinks listed.
+        :param or_tags: A list of tags that the TagLinks must be associated with at least one.
```
Comment (mrtizmoatwork) [inline src/polyswarm_api/api.py]: change to "A list of tags where the ..."

File: src/polyswarm_api/api.py [hunk 3]
```diff
@@ -349,6 +349,18 @@ def tag_link_update(self, sha256, tags=None, families=None, remove=False):
...
+        :param or_families: A list of families that the TagLinks must be associated with at least one.
```
Comment (mrtizmoatwork) [inline src/polyswarm_api/api.py]: change to "A list of families where the ..."

File: src/polyswarm_api/api.py (hunk 2 above)
Comment (sbnetops) [reply src/polyswarm_api/api.py]: But it is `a list of tags`, why should I remove that?

File: src/polyswarm_api/api.py (hunk 3 above)
Comment (sbnetops) [reply src/polyswarm_api/api.py]: Similarly here.

File: src/polyswarm_api/api.py (hunk 2 above)
Comment (mrtizmoatwork) [reply src/polyswarm_api/api.py]: Not remove, just change the word "that" to "where".

File: src/polyswarm_api/api.py (hunk 3 above)
Comment (mrtizmoatwork) [reply src/polyswarm_api/api.py]: same here

File: src/polyswarm_api/api.py [hunk 4]
```diff
@@ -354,8 +354,8 @@ def tag_link_list(self, tags=None, families=None, or_tags=None, or_families=None
         Fetch all existing TagLinks for the provided tags.
         :param tags: A list of tags that must be associated with the TagLinks listed.
         :param families: A list of families that must be associated with the TagLinks listed.
-        :param or_tags: A list of tags that the TagLinks must be associated with at least one.
-        :param or_families: A list of families that the TagLinks must be associated with at least one.
+        :param or_tags: A list of tags where the TagLinks must be associated with at least one.
+        :param or_families: A list of families where the the TagLinks must be associated with at least one.
```
Comment (mrtizmoatwork) [inline src/polyswarm_api/api.py]: remove duplicate "the"

Comment (sbnetops) [reply src/polyswarm_api/api.py]: done :grimacing:

**Pull request merged**